src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -355,6 +367,9 @@ private void UpdateWorkingDirectory()
                     return;
                 }
 
+                // save current solution directory so that we don't execute init scripts for the same solution again.
+                _currentSolutionDirectory = GetSolutionDirectory();
```

Comment:
Would it make more sense to compare the current and previous directories here inside the lock and noop if they were already run?  With the current pattern it looks like all (one) callers need to care about this.